### PR TITLE
[FEC-665] Add stories to EzIcon and update documentation

### PR DESCRIPTION
**Pull request by noranda**
## What did we change?
- [docs: add stories to EzIcon and update documentation](https://github.com/ezcater/recipe/commit/07c7c23027685a578e90b6f44a5d91934e5f9d63)

## Why are we doing this?
This is part of an effort to switch our doc site from Gatbsy to Storybook Docs.

## Screenshot(s) / Gif(s):
Screenshot 2023-10-20 at 10 34 45 AM
Screenshot 2023-10-20 at 10 35 00 AM
Screenshot 2023-10-20 at 10 35 14 AM

## Checklist
- [x] Create any remaining stories grouped as needed: `src/components/EzComponent/Documentation/Stories/Example1Stories.stories.tsx`
- [x] Create snapshot: `src/components/EzComponent/Documentation/EzComponent.snapshot.tsx`
- [x] Add any regression tests as a story: `src/components/EzComponent/Documentation/Stories/Regression.stories.tsx`
- [x] Move (or create) `EzComponent.snippets.tsx` into Documentation folder and make sure snippet matches default story
- [x] Delete old Gatsby files:
  - `src/components/EzComponent/EzComponent.md`
  - `src/components/EzComponent/EzComponent.preview.md`
  - `src/components/EzComponent/__tests__/EzComponent.test.md`
- [x] Test all the things:
  - [x] All control props work as expected
  - [x] All "Show Code" displays correctly
  - [x] Playroom works for docs and all stories
  - [x] Regression stories only show up in development mode: `STORYBOOK_INCLUDE_REGRESSION_STORIES=1 yarn storybook`
  - [x] Component library preview and link works
  - [x] Related component links work (if already in Storybook Docs)
  - [x] Add any changesets (only if there are recipe library changes)
  - [x] Git commits are organized
  - [x] `yarn lint` passes
  - [x] `yarn type-check` passes
  - [x] `yarn test` passes
  - [x] All chromatic tests pass/approved

### CODE REVIEW

**Review by pgrippi — APPROVED** [review]

File: packages/recipe/src/components/EzIcon/Documentation/Stories/Icon.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            </EzPage>
+          );
+        })()}
+      `,
+    },
+  },
+};
+
+const SVGIconJSX = dedent`
+  const starIcon = (
```
Comment (danidewitt) [inline packages/recipe/src/components/EzIcon/Documentation/Stories/Icon.stories.tsx:77]: worth pulling out as a variable?

File: packages/recipe/src/components/EzIcon/Documentation/Stories/Size.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+export const SizeInherit: Story = {
+  args: {
+    ...Default.args,
+    title: 'Example icon with inherited size',
+    size: 'inherit',
+  },
+  decorators: [
+    StoryFn => (
+      <div style={{fontSize: '5em'}}>
```
Comment (danidewitt) [inline packages/recipe/src/components/EzIcon/Documentation/Stories/Size.stories.tsx:100]: can we avoid inline styles? would using mui box be better? (same below)

**Review by danidewitt — DISMISSED** [review]
just a couple very minor comments

File: packages/recipe/src/components/EzIcon/Documentation/Stories/Size.stories.tsx (hunk 2 above)
Comment (noranda) [reply packages/recipe/src/components/EzIcon/Documentation/Stories/Size.stories.tsx:100]: This doesn't show up in our documentation as it's just a decorator, but I'll switch it over anyways.

**Review by pgrippi — DISMISSED** [review]

File: packages/recipe/src/components/EzIcon/Documentation/Stories/Size.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      <Box fontSize="5em">
+        <StoryFn />
+      </Box>
+    ),
+  ],
+  parameters: {
+    docs: {
+      source: {
+        code: dedent`
+              <div style={{fontSize: '5em'}}>
```
Comment (danidewitt) [inline packages/recipe/src/components/EzIcon/Documentation/Stories/Size.stories.tsx:110]: same thing with box

File: packages/recipe/src/components/EzIcon/Documentation/Stories/Size.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      },
+    },
+    playroom: {
+      code: dedent`
+        {(() => {
+          const {faCoffee} = require('@fortawesome/free-solid-svg-icons/faCoffee');
+      
+          return (
+            <EzPage>
+              <div style={{fontSize: '5em'}}>
```
Comment (danidewitt) [inline packages/recipe/src/components/EzIcon/Documentation/Stories/Size.stories.tsx:125]: same thing with box

**Review by danidewitt — APPROVED** [review]

**Review by pgrippi — APPROVED** [review]

**Pull request merged**